pyramid/session.py

```diff
@@ -626,3 +637,46 @@ def SignedCookieSessionFactory(
         reissue_time=reissue_time,
         set_on_exception=set_on_exception,
     )
+
+class BlowfishPickleSerializer(object):
+    """ Encrypting Webob cookie serializer.
+
+    Uses the pickle protocol to dump Python data to bytes, then encrypts
+    the value symmetrically using the Blowfish algorithm.
+
+    To encrypt sessions, pass an instance of this class as the ``serializer``
+    argument to :class:`BaseCookieSessionFactory`.
+
+    .. note::
+
+       Before using this serializer, install ``pyramid[encrypted_cookies]`` to
+       get the necessary dependency (``pycrypto``).
+    """
+    def __init__(self, secret):
```

Comment:
There is no signing of the content since it's using the `BaseCookieSessionFactory`. The signing needs to be done with a separate secret from the encryption key. There is also no salt to prevent dual-use of secrets in the application. See the `SignedCookieSessionFactory` for an example of the salt.